src/main/java/io/github/genomicdatainfrastructure/daam/mappers/RemsApplicationMapper.java

```diff
@@ -6,157 +6,241 @@
-                .lastActivity(remsApplication.getApplicationLastActivity())
-                .events(toRetrievedApplicationEvents(remsApplication.getApplicationEvents()))
-                .roles(remsApplication.getApplicationRoles().stream().toList())
-                .attachments(toRetrievedApplicationAttachments(remsApplication
-                        .getApplicationAttachments()))
-                .licenses(toRetrievedApplicationLicences(remsApplication.getApplicationLicenses()))
-                .createdAt(remsApplication.getApplicationCreated())
-                .state(RetrievedApplication.StateEnum.fromString(remsApplication
-                        .getApplicationState()
-                        .value()))
-                .modifiedAt(remsApplication.getApplicationModified())
+                .id(application.getApplicationId())
+                .externalId(application.getApplicationExternalId())
+                .description(application.getApplicationDescription())
+                .generatedExternalId(application.getApplicationGeneratedExternalId())
+                .lastActivity(application.getApplicationLastActivity())
+                .roles(application.getApplicationRoles())
+                .createdAt(application.getApplicationCreated())
+                .modifiedAt(application.getApplicationModified())
+                .workflow(toWorkflow(application))
+                .applicant(toApplicant(application))
+                .members(toMembers(application))
+                .invitedMembers(toInvitedMembers(application))
+                .datasets(toDatasets(application))
+                .forms(toForms(application))
+                .permissions(toPermissions(application))
+                .events(toEvents(application))
+                .attachments(toAttachments(application))
+                .licenses(toLicences(application))
+                .state(toState(application))
                 .build();
     }
 
-    private static RetrievedApplicationWorkflow toRetrievedApplicationWorkflow(
-            Response10953Workflow remsWorkflow) {
-        return new RetrievedApplicationWorkflow(remsWorkflow.getWorkflowId(),
-                remsWorkflow.getWorkflowType());
+    private RetrievedApplicationWorkflow toWorkflow(Application application) {
+        Optional<Response10953Workflow> workflow = Optional.ofNullable(application
+                .getApplicationWorkflow());
+
+        return new RetrievedApplicationWorkflow(workflow.map(Response10953Workflow::getWorkflowId)
+                .orElse(null),
+                workflow.map(Response10953Workflow::getWorkflowType).orElse(null));
     }
 
-    private static RetrievedApplicationApplicant toRetrievedApplicationApplicant(
-            UserWithAttributes remsApplicant) {
+    private RetrievedApplicationApplicant toApplicant(Application application) {
+        Optional<UserWithAttributes> applicant = Optional.ofNullable(application
```

Comment:
suggestion: to improve readability, you can simply assign to `var` in all the cases.